app/scripts/proactive/view/dom.js

```diff
@@ -428,6 +428,11 @@ define(
                 return;
             }
 
+            for( var key in jobVariables){
+                if(jobVariables[key].Model.indexOf('$') !== -1){
```

Comment:
This will clash too much with models which can contain $ (for example regexp).

The correct way is to build a set or list of possible variable references by iterating on the keys first and adding in this set/list, for each key:

$key
${key}